File: frontend/src/types/report.ts
// 报告相关类型

export interface TestReport {
  id: number
  planId: number
  planName: string
  projectName: string
  totalCount: number
  passedCount: number
  failedCount: number
  skippedCount: number
  errorCount: number
  passRate: number    // 0-100
  duration: number    // 总耗时（ms）
  startTime: string
  endTime: string
  caseResults: CaseResult[]
  trendData?: TrendDataPoint[]
}

export interface CaseResult {
  caseId: number
  caseTitle: string
  priority: string
  module: string
  status: 'passed' | 'failed' | 'skipped' | 'error'
  duration: number
  errorMessage?: string
  screenshots: string[]
  steps: ExecutionStepBrief[]
}

export interface ExecutionStepBrief {
  seq: number
  action: string
  target: string
  expected: string
  actual: string
  status: 'passed' | 'failed'
  screenshot?: string
}

export interface TrendDataPoint {
  date: string
  passRate: number
  totalCount: number
  failedCount: number
}

File: backend/app/models/report.py
"""报告模型"""

from datetime import datetime
from sqlalchemy import String, Integer, Text, JSON, DateTime, Float, ForeignKey, func
from sqlalchemy.orm import Mapped, mapped_column
from app.database import Base


class TestReport(Base):
    __tablename__ = "reports"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    plan_id: Mapped[int] = mapped_column(Integer, nullable=False, index=True)
    plan_name: Mapped[str] = mapped_column(String(256), nullable=False)
    project_id: Mapped[int] = mapped_column(Integer, nullable=False)
    project_name: Mapped[str] = mapped_column(String(128), nullable=False)
    total_count: Mapped[int] = mapped_column(Integer, default=0)
    passed_count: Mapped[int] = mapped_column(Integer, default=0)
    failed_count: Mapped[int] = mapped_column(Integer, default=0)
    skipped_count: Mapped[int] = mapped_column(Integer, default=0)
    error_count: Mapped[int] = mapped_column(Integer, default=0)
    pass_rate: Mapped[float] = mapped_column(Float, default=0)
    duration: Mapped[float] = mapped_column(Float, default=0)
    case_results: Mapped[list] = mapped_column(JSON, default=list, comment="各用例结果明细")
    trend_data: Mapped[list] = mapped_column(JSON, nullable=True, comment="近 7 天趋势")
    start_time: Mapped[datetime] = mapped_column(DateTime, nullable=False)
    end_time: Mapped[datetime] = mapped_column(DateTime, nullable=False)
    created_at: Mapped[datetime] = mapped_column(DateTime, server_default=func.now())


class CaseResult(Base):
    """已废弃 — 用例结果直接以 JSON 存储在 reports.case_results 中"""
    __tablename__ = "case_results"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    report_id: Mapped[int] = mapped_column(Integer, ForeignKey("reports.id"), nullable=False)
    case_id: Mapped[int] = mapped_column(Integer, nullable=False)
    case_title: Mapped[str] = mapped_column(String(256), nullable=False)
    priority: Mapped[str] = mapped_column(String(8), nullable=False)
    module: Mapped[str] = mapped_column(String(128), nullable=True)
    status: Mapped[str] = mapped_column(String(32), nullable=False)
    duration: Mapped[float] = mapped_column(Float, default=0)
    error_message: Mapped[str] = mapped_column(Text, nullable=True)
    screenshots: Mapped[list] = mapped_column(JSON, default=list)
    steps: Mapped[list] = mapped_column(JSON, default=list)
